src/gen_lockstep.erl

```diff
@@ -372,17 +372,31 @@ parse_uri(Url) ->
         Uri -> Uri
     end.
 
-ssl_upgrade(https, Sock) ->
-    case ssl:connect(Sock, []) of
+ssl_upgrade(https, Hostname, Sock) ->
+    SslOptions = ssl_opts(Hostname),
+    case ssl:connect(Sock, SslOptions) of
         {ok, SslSock} ->
             {ok, ssl, SslSock};
         Err ->
             Err
     end;
 
-ssl_upgrade(http, Sock) ->
+ssl_upgrade(http, _Hostname, Sock) ->
     {ok, gen_tcp, Sock}.
 
+ssl_opts(Hostname) ->
+    case application:get_env(lockstep, verify_ssl, false) of
+        true ->
+            VerifyFun = {fun ssl_verify_hostname:verify_fun/3, [{check_hostname, Hostname}]},
+            CACerts = certifi:cacerts(),
+            [{verify, verify_peer},
+             {depth, 2},
+             {cacerts, CACerts},
+             {verify_fun, VerifyFun}];
```

Comment:
You could pass in `{server_name_indication, Hostname}` so we send SNI headers as we update from TCP to SSL.